[UI]: add a version tag to CatalogCard component

SAHU-01 opened the pull request:
**Notes for Reviewers**

This PR fixes #688

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

<!--
Thank you for contributing to Meshery!

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR.
3. Sign your commits

By following the community's contribution conventions upfront, the review process will
be accelerated and your PR merged more quickly.
-->

sudhanshutech commented on src/custom/CatalogCard/CatalogCard.tsx:
```diff
@@ -38,8 +38,21 @@ type CatalogCardProps = {
+  position: 'absolute',
+  top: '40px',
+  right: '8px',
+  backgroundColor: theme.palette.primary.main,
```
use color from palette @SAHU-01

sudhanshutech commented on src/custom/CatalogCard/CatalogCard.tsx:
```diff
@@ -38,8 +38,21 @@ type CatalogCardProps = {
+  position: 'absolute',
+  bottom: '65px',
+  left: '18px',
+  backgroundColor: theme.palette.grey[500],
```
not these, there is tokens we use you will get those in palette.js file.

SAHU-01 replied:
> not these, there is tokens we use you will get those in palette.js file.

Thank you for sharing this, I'll make sure to take a look at it and modify this.

sudhanshutech commented:
@dottharun

SAHU-01 commented:
@sudhanshutech based on the Wednesday call feedback that I received,  I need to add the flipable thingy and also centralize the catalog card and add on flip all the details that were available on meshery cloud. I'll be done with this over the weekend and will make sure to notify you.

sudhanshutech commented:
oh its good. I was tagging @dottharun to confirm is he has some things to confirm here?

dottharun commented:
@SAHU-01 Does this support dark mode, and can you provide screenshots if you can?

SAHU-01 commented:
@dottharun the tag color doesn't change with dark or light mode switch as of now if you want me to enable that, do let me know
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/8829e58a-1ae6-4d2b-ae97-fd4dd40df56d">
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/3b6a8bbe-f93f-46c8-a98b-0949f9420588">

dottharun commented:
@SAHU-01 Yes, please use the same shades as the footer.

SAHU-01 commented:
@dottharun I did add the version background based on the color palette used for metrics container and it looks something like this:
for dark theme:
<img width="273" alt="image" src="https://github.com/user-attachments/assets/0af6b55b-230c-4b23-ae5b-ca1b81e7b432">
for light theme:
<img width="273" alt="image" src="https://github.com/user-attachments/assets/88791d23-b5c0-43cc-8e24-f070357e7cda">

SAHU-01 commented:
@sudhanshutech, I have a couple of questions here:

1. should I move all styled components like this to `src>custom>CatalogCard>style.css`:
```
const BackFaceContent = styled(Box)(({ theme }) => ({
  position: 'absolute',
  background: `linear-gradient(to bottom right, black 40%, ${theme.palette.background.brand?.default})`,
  width: '100%',
  top: 0,
  left: 0,
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'left',
  padding: '16px',
  boxShadow: `2px 2px 3px 0px black`,
  borderRadius: '1rem'
}));
```
2. As discussed earlier for lining issues I've added a new interface do let em know if that requires changes.
3. @dottharun had suggested adding version background based on the color palette used for `metrics container` please take a look at it and let me know if that suits or we could opt for a different color palette
4. The linear gradient is not similar to the one on [Catalog Page](https://meshery.layer5.io/catalog) I was unable to find a figma for the flip catalog card, do let me know if that needs to be changed, I would need the exact hex codes though.

SAHU-01 commented:
Please find the demo screen here:

https://github.com/user-attachments/assets/c8849249-b816-4044-b9ea-2906b296187f

sudhanshutech commented:
@SAHU-01 yes the styles should be in separate style folder

dottharun commented:
@SAHU-01 If we are using the flip card and showing the version only on the backside, just use the same shades as in https://meshery.layer5.io/.

sudhanshutech commented:
any more help needed @SAHU-01 ?

SAHU-01 commented:
@sudhanshutech yes the figma design? Does the gradient and padding look good to you in the recording? If yes I'll push the style file and tag theme changes now?

sudhanshutech commented:
the color of metrics in from of card could be shade of white for dark mode. See which looks good from using the token.

sudhanshutech commented:
@SAHU-01 have gone through the card here in cloud,: https://meshery.layer5.io/catalog?

sudhanshutech commented:
@dottharun @vishalvivekm @captain-Akshay do we have figma design for this card?

SAHU-01 commented:
@dottharun, I've removed the flip logic from this PR and I've added the `versionTag` and the `versionTagPosition` props with proper styling as per your suggestion.
@sudhanshutech I'll be sending the other PR with centralized sistent logic soon.
Currently this is how it looks:

https://github.com/user-attachments/assets/3af4314b-855a-4d43-8b98-34d6e4750eeb
